tests/lib/ble_gq/src/unity_test.c

```diff
@@ -0,0 +1,319 @@
+/*
+ * Copyright (c) 2025 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+#include <unity.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <strings.h>
+#include <errno.h>
+
+#include <ble_qwr.h>
+#include <ble_gq.h>
+#include <zephyr/sys/slist.h>
+#include "cmock_ble_gattc.h"
+#include "cmock_ble_gatts.h"
+
+#define MAX_CONNS 2
+#define BLE_GQ_REQ_UNDEFINED 7
+#define BLE_GQ_QUEUE_SIZE 8
+#define BLE_GQ_HEAP_SIZE 1024
+
+BLE_GQ_CUSTOM_DEF(ble_gq, MAX_CONNS, BLE_GQ_HEAP_SIZE, MAX_CONNS * BLE_GQ_QUEUE_SIZE);
+static uint16_t glob_conn_handle;
+static uint32_t glob_error;
+
+static void ble_gq_error_handler(uint16_t conn_handle, uint32_t err, void *ctx)
+{
+	glob_conn_handle = conn_handle;
+	glob_error = err;
+}
+
+void test_ble_gq_item_add_efault(void)
+{
+	struct ble_gq_req req = {0};
+	int ret;
+
+	ret = ble_gq_item_add(NULL, &req, 1);
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+
+	ret = ble_gq_item_add(&ble_gq, NULL, 1);
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_ble_gq_item_add_einval(void)
+{
+	struct ble_gq_req req = {.type = BLE_GQ_REQ_NUM};
+	int ret;
+
+	ret = ble_gq_item_add(&ble_gq, &req, 0);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+
+	ret = ble_gq_item_add(&ble_gq, &req, 99);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+}
+
+void test_ble_gq_item_add_req_gatt_read(void)
+{
+	uint16_t conn_handle = 0;
+	int ret;
+	struct ble_gq_req req = {
+		.type = BLE_GQ_REQ_GATTC_READ,
+		.error_handler = {
+			.cb = ble_gq_error_handler,
+			.ctx = NULL
+		},
+		.gattc_read = {
+			.handle = 0,
+			.offset = 0
+		}
+	};
+	ble_gq.conn_handles[0] = 0;
+
+	__cmock_sd_ble_gattc_read_ExpectAndReturn(conn_handle, req.gattc_read.handle,
+			req.gattc_read.offset, NRF_ERROR_BUSY);
+	__cmock_sd_ble_gattc_read_ExpectAndReturn(conn_handle, req.gattc_read.handle,
+			req.gattc_read.offset, NRF_ERROR_INVALID_STATE);
+
+	ret = ble_gq_item_add(&ble_gq, &req, conn_handle);
+	TEST_ASSERT_EQUAL(0, ret);
+	TEST_ASSERT_EQUAL(conn_handle, glob_conn_handle);
+	TEST_ASSERT_EQUAL(NRF_ERROR_INVALID_STATE, glob_error);
```

Comment:
Note for later, we have a module returning both errnos and nrf_errors. This is not good! @anhmolt @MirkoCovizzi